pkg/cloud-init/cloud-init.go

```diff
@@ -108,6 +112,16 @@ type DeviceData struct {
 	Tags        []string           `json:"tags"`
 }
 
+type Chpasswd struct {
+	Expire *bool `json:"expire,omitempty"`
+}
+
+type UserData struct {
+	Password string `json:"password"`
+	Chpasswd Chpasswd
```

Comment:
nit: I don't think `Password` and `Chpasswd` fields are needed, since we want to unmarshal only the `Hostname`

Reply:
My personal preference would to include only what is needed. But I do not insist, so up to you :)

Reply:
... and in fact this struct is used only in one place as I can see. What do you think about defining an anonymous struct in-place then? E.g.

```go
u := struct {
    Hostname string `json:"hostname"`
} {}
```

Reply:
I've included hostname only :)
I like your thinking, but my main concern is that in the future someone will also need to marshall the user data and will re-create the same structure. Maybe I'm too paranoid, but I prefer it would be in a visible place for possible future uses

Reply:
I am just a follower of the [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) principle... but again, I do not insist :)

Reply:
I'm a big believer in YAGNI :) but in this case I don't believe we gain much value from not exposing it into a local scope